<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>RNA-seq / Assemble reads into transcripts using Cufflinks</h2>

<h3>Description</h3>

<p>
This tool assembles RNA-seq reads into transcripts using version Cufflinks 2.2.1.
</p>

<h3>Parameters</h3>

<ul>
<li>Chromosome names in my BAM file look like (chr1, 1) [1]</li>
<li>Enable multi-mapped read correction (yes,no) [no]</li>
<li>Correct for sequence-specific bias (yes, no) [no]</li>
<li>Estimate expression of known isoforms, don't assemble novel ones (yes, no) [no]</li>
<li>Do reference annotation based transcript assembly (yes, no)[no]</li>
<li>Reference organism</li>
</ul>

<h3>Details</h3>

<p>
Given aligned RNA-Seq reads, Cufflinks assembles the alignments into a parsimonious set of transcripts. 
It then estimates the relative abundances of these transcripts based on how many reads support each one, 
taking into account biases in library preparation protocols. It is recommended to use the TopHat aligner 
to map your reads to the reference genome. You can merge the resulting GTFs from several samples to 
one using the Cuffmerge tool, and use it in differential expression analysis using Cuffdiff.
</p>
<p>
Cufflinks can detect sequence-specific bias and correct for it in abundance estimation. </p>
<p>
By default, Cufflinks will uniformly divide each multi-mapped read to all of the positions it maps to. If multi-mapped read correction is enabled, Cufflinks will re-estimate the transcript abundances dividing each multi-mapped read probabilistically based on the initial abundance estimation, the inferred fragment length and fragment bias, if bias correction is enabled.
</p>
<p>
Cufflinks can perform reference annotation based transcript assembly (RABT) where reference transcripts will be tiled with faux-reads to provide 
additional information in assembly. If you enable RABT, the output will include all reference transcripts as well as any novel genes and isoforms that are assembled.
Please note that RABT can produce false positive novel transcripts when used with complete annotations. Therefore it is recommended only for organisms where deep annotations do not already exist.
</p>

<h3>Output</h3>

<p>The analysis output consists of the following:</p>
<ul>
<li>transcripts.gtf: This GTF file contains Cufflinks' assembled isoforms. The first 7 columns are standard GTF, and the last column contains attributes, some of which are also standardized ("gene_id", and "transcript_id"). There is one GTF record per row, and each record represents either a transcript or an exon within a transcript.</li>
<li>genes.fpkm_tracking.tsv: This file contains the estimated gene-level expression values in the generic FPKM Tracking Format. Note, however that as there is only one sample, the "q" format is not used.</li>  
<li>isoforms.fpkm_tracking.tsv: This file contains the estimated isoform-level expression values in the generic FPKM Tracking Format. Note, however that as there is only one sample, the "q" format is not used.</li> 
<li>skipped.gtf: This file contains the skipped loci.</li>
</ul>
</p>


<h3>References</h3>

<p>This tool uses the <a href="http://cole-trapnell-lab.github.io/cufflinks/cufflinks/index.html">Cufflinks</a> package for statistical analysis. Please cite the following article:</p>

<p>Trapnell C, Williams BA, Pertea G, Mortazavi A, Kwan G, van Baren MJ, Salzberg
SL, Wold BJ, Pachter L. <a href="http://www.ncbi.nlm.nih.gov/pubmed/20436464">Transcript assembly and quantification by RNA-Seq reveals unannotated transcripts and isoform switching during cell differentiation. Nat Biotechnol. 2010 May;28(5):511-5.</a>
</p> 
<p>
If you use RABT, please cite 
</p>
<p>Adam Roberts, Harold Pimentel, Cole Trapnell, Lior Pachter. <a href="http://bioinformatics.oxfordjournals.org/content/27/17/2325">Identification of novel transcripts in annotated genomes using RNA-seq. Bioinformatics 2011. 27 (17): 2325-2329.</a>
</p>
</body>

