<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>RNA-seq / Compare assembled transcripts to reference annotation using Cuffcompare</h2>

<h3>Description</h3>

<p>
Cuffcompare allows you to compare the transfrags you assemble to a reference annotation.
</p>

<h3>Parameters</h3>

<ul>
<li>Chromosome names in my GTF files look like (chr1, 1) [1]</li>
<li>Ignore non-overlapping reference transcripts (yes, no) [no]</li>
<li>Reference organism</li>
</ul>

<h3>Details</h3>

<p>
Cufflinks GTF from the sample is matched against the reference GTF, and sample isoforms are tagged as overlapping, 
matching, or novel where appropriate. You can supply your own reference GTF or use one of the provided ones.
</p>


<h3>Output</h3>

<p>
<p>The analysis output consists of the following files:</p>
<ul>
<li>cuffcmp.refmap.tsv</li>
<li>cuffcmp.tmap.tsv</li>
<li>cuffcmp.combined.gtf</li>
<li>cuffcmp.loci.tsv</li>
<li>cuffcmp.stats.txt</li>
<li>cuffcmp.tracking.tsv</li>
</ul>
Please see <a href="http://cole-trapnell-lab.github.io/cufflinks/cuffcompare/">Cufflinks documentation</a> for details of output files.
</p>


<h3>References</h3>

<p>This tool uses the <a href="http://cole-trapnell-lab.github.io/cufflinks/">Cufflinks</a> package for statistical analysis. Please read the following article for more detailed information:</p>

<p>Trapnell C, Williams BA, Pertea G, Mortazavi A, Kwan G, van Baren MJ, Salzberg
SL, Wold BJ, Pachter L. <a href="http://www.ncbi.nlm.nih.gov/pubmed/20436464">Transcript assembly and quantification by RNA-Seq reveals unannotated transcripts and isoform switching during cell differentiation. Nat Biotechnol. 2010 May;28(5):511-5.</a>
</p> 

</body>

