<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Variants / Tabulate pileup metrics for inferring contamination with GATK4</h2>

<h3>Description</h3>

<p>
Summarizes counts of reads that support reference, alternate and other alleles for given sites.
</p>

<h3>Parameters</h3>

<ul>
<li>Reference sequence () [human]</li>
<li>Chromosome names in my BAM file look like (1, chr1) [chr1]</li><li>Reference sequence () [human]</li>
<li>Chromosome names in my BAM file look like (1, chr1) [chr1]</li>
<li>Genomic intervals /String []</li>
<li>Interval padding (Integer) [0]</li>
<li>Use Variants VCF also for intervals (yes, no) [no]</li>
</ul>


<h3>Details</h3>

<p>
Summarizes counts of reads that support reference, alternate and other alleles for given sites. 
Results can be used with CalculateContamination. 
</p>
<p>
</p>
The tool requires a common germline variant sites VCF, e.g. derived from the gnomAD resource, 
with population allele frequencies (AF) in the INFO field. This resource must contain only 
biallelic SNPs and can be an eight-column sites-only VCF. The tool ignores the filter status 
of the variant calls in this germline resource. 
<p>
Based on GATK4 GetPileupSummaries tool. For more detailed information, see <a href = "https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_hellbender_tools_walkers_contamination_GetPileupSummaries.php">GATK documentation</a>,
</p>

<h3>Output</h3>

<p>
<ul>
<li></li>
</ul>
</p>


<h3>References</h3>

<p>This tool is based on the <a href = "https://software.broadinstitute.org/gatk/gatk4">GATK4 package</a>.</p>
<p>
For instructions on citing GATK4, please see <a href = "https://software.broadinstitute.org/gatk/documentation/article?id=11027">GATK FAQ</a> 
</p>

</body>

