<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Remove chimeric sequences</h2>

<h3>Description</h3>

<p>
Given a fasta-formatted alignment file, removes chimeric sequences.
</p>

<h3>Parameters</h3>

<ul>
<li>Method (vsearch, uchime) [vsearch]</li>
<li>Reference (bacterial, full, none) [bacterial]</li>
<li>Dereplicate (false, true) [false]</li>
</ul>

<h3>Details</h3>
<p>
Removes chimeric sequences (artifacts which contain parts of several sequences) from an alignment using the vsearch or uchime method. The vsearch method is faster.
</p>
<p>
You can select de novo or reference based chimera detection. As a reference you can use the bacterial subset of the 16S rRNA Silva gold reference set. For the uchime method also the full 16S rRNA Silva gold reference set is available. 
When using the de novo approach, the tool checks for chimeras using the more abundant sequences in your data as the reference.
</p>
<p>
The dereplicate parameter is by default false, so when a sequence is detected as chimeric in one sample it is assigned chimeric all the samples. If you want to limit the assignment to the original sample, set the dereplicate parameter to true.  
</p>
<p>
This tool is based on the <a href="http://www.mothur.org/wiki/Chimera.vsearch">Chimera.vsearch</a>, <a href="http://www.mothur.org/wiki/Chimera.uchime">Chimera.uchime</a>, and <a href="http://www.mothur.org/wiki/Remove.seqs">Remove.seqs</a> commands of the Mothur package.
</p>

<h3>Output</h3>

<p>The analysis output consists of the following:</p>
<ul>
<li>chimeras-removed.fasta: Alignment with chimeras removed</li> 
<li>chimeras.removed.count_table: Count table with chimeras removed</li>
<li>chimeras-removed-summary.tsv: Summary statistics for the alignment </li> 
</ul>
</p>


<h3>References</h3>
<p>Schloss PD, Westcott SL, Ryabin T, Hall JR, Hartmann M, Hollister EB, Lesniewski RA, Oakley BB, Parks DH, Robinson CJ, Sahl JW, Stres B, Thallinger GG, Van Horn DJ, Weber CF.
<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2786419/">Introducing mothur: Open-source, platform-independent, community-supported software for describing and comparing microbial communities. Appl Environ Microbiol, 2009. 75(23):7537-41.</a> 
</p>

</body>

