<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Remove chimeric sequences</h2>

<h3>Description</h3>

<p>
Given a fasta-formatted alignment file, removes chimeric sequences.
</p>

<h3>Parameters</h3>

<p>Dereplicate (false, true) [false]</p>
<p>Reference (bacterial, full, none) [bacterial]</p>

<h3>Details</h3>
<p>
Removes chimeric sequences from an alignment using the uchime method and the 16S rRNA Silva gold reference set. Chimeric sequences are artifacts which contain parts of several sequences.
</p>
<p>
The dereplicate parameter can be used when checking for chimeras by group. 
If the dereplicate parameter is false (default), then if one group finds the sequence to be chimeric, 
all groups find it to be chimeric. 
</p>
<p>
You can choose whether you want to use the full Silva reference set, or just the bacterial sequences. 
If you choose none, the tool checks for chimeras using the more abundant sequences as the reference.
</p>
<p>
This tool is based on the <a href="http://www.mothur.org/wiki/Chimera.uchime">Chimera.uchime</a> and <a href="http://www.mothur.org/wiki/Remove.seqs">Remove.seqs</a> commands of the Mothur package.
</p>

<h3>Output</h3>

<p>The analysis output consists of the following:</p>
<ul>
<li>chimeras-removed.fasta: Alignment with chimeras removed</li> 
<li>chimeras.removed.count_table: Count table with chimeras removed</li>
<li>chimeras-removed-summary.tsv: Summary statistics for the alignment </li> 
</ul>
</p>


<h3>References</h3>
<p>Schloss PD, Westcott SL, Ryabin T, Hall JR, Hartmann M, Hollister EB, Lesniewski RA, Oakley BB, Parks DH, Robinson CJ, Sahl JW, Stres B, Thallinger GG, Van Horn DJ, Weber CF.
<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2786419/">Introducing mothur: Open-source, platform-independent, community-supported software for describing and comparing microbial communities. Appl Environ Microbiol, 2009. 75(23):7537-41.</a> 
</p>

</body>

