<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Analysing differential expression with RNA-seq data</h2>

<p>
We have gathered all of our learning material for RNA-seq data analysis to this page to help you out to get started!
<br>
If RNA-seq is a whole new topic to you, we would suggest checking out the
 <a href= "https://www.youtube.com/watch?v=Cb37UU7ahe0&t=4s&list=PLjiXAZO27elABzLA0aHKS9chVA2TldoPF&index=6"> introduction to RNA-seq webinar</a> (1h), 
which introduces the steps of differential expression analysis -this video covers also the 101 of using Chipster.
<br><br>
After the webinar, you might like to try the analysis tools yourself -for that we suggest going through our course exercises
(listed below). Open the excercises pdf document, log in to Chipster, and just follow the instructions!
<br><br>
Some parts of the analysis can raise more questions -we have tried to clarify those steps with more 
detailed instructions and tutorials. Please find the links to those on this page!
</p>
<h3>Course material</h3>
<p>
Please find below the links to our RNA-seq course materials.<br>
Course example sessions are available in Chipster (File -> Open example session -> "course_RNAseq..."). 
<ul>
<li>Introduction to RNA-seq data analysis with Chipster <a href= "https://kannu.csc.fi/index.php/s/zqHXWdr32yOA5xo">course slides and exercises</a></li>
<li>Introduction to RNA-seq data analysis, command line tools + R/Bioconductor  <a href= "https://kannu.csc.fi/index.php/s/4oGtgfglXOOjCzI">course slides, exercises and some test data</a></li>
</ul>
</p>

<h3>Tutorial videos</h3>

We have an <a href= "https://www.youtube.com/playlist?list=PLjiXAZO27elABzLA0aHKS9chVA2TldoPF">RNA-seq playlist</a> in our YouTube channel. 
This list includes tutorials for some common analysis needs:
<ul>
<li>Beginner's introduction to RNA-seq data analysis <a href= "https://www.youtube.com/watch?v=Cb37UU7ahe0&t=4s&list=PLjiXAZO27elABzLA0aHKS9chVA2TldoPF&index=6"> Webinar</a> (1h)</li>
<li>Differential expression analysis tools for RNA-seq <a href= "https://www.youtube.com/watch?v=kPDy4FYF1yY&t=16s&list=PLjiXAZO27elABzLA0aHKS9chVA2TldoPF&index=1">tutorial video</a> (3 min)</li>
<li>EdgeR for multivariate experiments<a href= "https://www.youtube.com/watch?v=gJnu3xUQ7ww&list=PLjiXAZO27elABzLA0aHKS9chVA2TldoPF&index=3"> tutorial video</a> (6 min)</li>
<li>Using the "nested" option in multivariate edgeR <a href= "https://www.youtube.com/watch?v=82G_yMkPue4&t=11s&list=PLjiXAZO27elABzLA0aHKS9chVA2TldoPF&index=2">tutorial video</a> (4 min)</li>
<!--  <li>Upper-quartile normalization (yes, no) [yes]</li>-->
</ul>
</p>
<h3>Getting started with Chipster</h3>
<p>
<ul>
<li>Do you have a user account? <a href= "http://chipster.csc.fi/access.shtml">Instructions for getting access</a> </li>
<li>Getting started <a href= "https://www.youtube.com/watch?v=mgK1eo1YW4I">tutorial video</a> (12 min)</li>
<li>Opening Chipster with Mac: <a href= "https://www.youtube.com/watch?v=4uAJJ3HEc4w">tutorial video</a> (2 min)</li>
<li>Saving and opening your sessions: <a href= "http://chipster.csc.fi/manual/sessions.html">Manual page</a> and <a href="https://www.youtube.com/watch?v=KOeSRSBy-LU">tutorial video</a> (2 min)</li>
<li>What tools are available? <a href= "http://chipster.csc.fi/manual/tools-ngs.html#RNA">List of tools</a></li>
</ul>
</p>

<h3>Importing your data to Chipster</h3>
<p>
<ul>
<li>Importing your files to Chipster: <a href= "http://chipster.csc.fi/manual/import-ngs.html">Manual</a></li>
<li>Have your data already in CSC's Taito cluster, how to run Chipster there? <a href= "https://www.youtube.com/watch?v=riipGwF-Qxs">Video tutorial</a> (12 min)</li>
</ul>
</p>

<h3>Analysis steps for finding differentially expressed genes</h3>

These steps take you from raw reads to differentially expressed genes (please find a figure showing the different steps and file formats at the end of the page):
<ol>

<li><b>Quality control of raw reads </b><br>
	Check the number, quality encoding, strandedness and inner distance of your reads, and inspect them for base quality, biases and adapters.<br>
	Tools under Quality control category (input FASTQ files):
	<ul> 
	<li><a href= "http://chipster.csc.fi/manual/fastqc.html">FastQC </a> is clear and fast.</li>
	<li><a href= "http://chipster.csc.fi/manual/prinseq-statistics.html">PRINSEQ</a></li>
	<li><a href= "http://chipster.csc.fi/manual/rseqc_infer_rnaseq_experiment.html"> RNA-seq strandedness inference and inner distance estimation using RseQC</a> 
	checks if the data was produced using a stranded protocol and what is the inner distance of paired reads. This information is important for the subsequent alignment and quantitation steps.</li>
	</ul>
</li>
<br>
<li><b>Preprocessing raw reads </b><br>
	If the reads contain low quality bases or adapter sequences, you might like to trim or filter them. 
	Note that when preprocessing paired end data, you need to give the two read files simultaneously to the analysis tool in order to preserve the order of read pairs.<br>
	Trimming or filtering is not absolutely necessary, because many bad quality reads are either removed or dealt with when aligning reads to the reference genome in the next step.
	This depends on the aligner used, as aligners differ in their ability to cope with mismatching bases. 
	<br>
	Tools under Preprocessing category (input FASTQ files):
   	<ul>
	<li><a href= "http://chipster.csc.fi/manual/trimmomatic.html">Trimmomatic</a> has several options for trimming and filtering and it is fast.</li>
	<li><a href= "http://chipster.csc.fi/manual/prinseq-trimmer.html">PRINSEQ trimming</a> is slower.</li>
	<li><a href= "http://chipster.csc.fi/manual/prinseq-filter.html">PRINSEQ filtering</a> is slower.</li>
	</ul>
<br>
<li><b>Alignment of reads to reference genome</b><br>
	 Chipster offers a large selection of reference genomes. If your reference genome is not available, you can import it as a fasta file and use the "own genome" version of the alignment tool. 
	 You can also ask us to add the genome in Chipster if it is publicly available and commonly used.<br> 
	 Note that some sequencing platforms generate <b>several FASTQ files per sample</b>. For example, Illumina NextSeq generates 8 files per sample for paired end data. 
	 In this case you need to first generate file name lists (one for read 1 files and another for read 2 files), and give those list files and all the FASTQ files as input for the aligner.
	 Please read the manual for the <a href= "http://chipster.csc.fi/manual/make_namelist.html">Utilities/Make a list of file names</a> tool for more info. <br>
	 Note that aligners need <b>strandedness / library type</b> information in order to align reads correctly. As there are two possible strandedness types 
	 and several kits for producing stranded sequencing libraries, we have made a <a href="http://chipster.csc.fi/manual/library-type-summary.html">summary of strandedness options and nomenclature</a>.
	 <br>
	 Tools under Alignment category, note that there are separate TopHat tools for single and paired end data, and for user-supplied reference genome (input FASTQ files):
	 <ul>
	 <li><a href= "http://chipster.csc.fi/manual/tophat2.html">TopHat for paired end reads</a> and other TopHat tools.</li>
	 <li>Coming soon: STAR and HISAT2 aligners</li>
	 </ul>
<br>
<li><b>Alignment level quality control: </b><br>	
	 Check what proportion of the reads mapped to exons, is the coverage uniform over transcript length, and whether novel splice sites were found.
	 Tools under Quality control category (input BAM files):
	 <ul> 
	 <li><a href= "http://chipster.csc.fi/manual/rseqc.html">RSeQC </a> 
	 <li><a href= "http://chipster.csc.fi/manual/picard-collectmultiplemetrics.html">CalculateMultipleMetrics (Picard)</a></li>
	 </ul>
<br>
<li><b>Quantitation (for BAM files): </b><br>	
	Next we want to count the reads per gene/transcript. The quantitation is done separately for each BAM file.
	At this step you also need to report the library type / strandedness -make sure you choose the correct parameter!
	<br>
	 Tools (under RNA-seq category): 
	 <a href= "http://chipster.csc.fi/manual/htseq-count.html">HTSeq </a>, 
	 <a href= "http://chipster.csc.fi/manual/dexseq-exoncount.html">DEXSeq </a></li>
<br>
<li><b>Define NGS experiment (for the count files): </b><br>	
	Now we need to combine all the count files (samplename.tsv) into a one table (ngs-data-table.tsv), 
	where rows represent genes and columns different samples. 
	For this, select all the blue count files and run the tool Utilities/Define NGS experiment.<br>
	This tool generates the table and a <b>phenodata file </b>. This is your way to describe your samples in Chipster:
	your next task is to fill in the "group" column and maybe add some more columns that describe which of your
	samples are "controls", "treatments", from same batch/patient etc. 
	You also want to make sure the "Description" column is, well, descriptive, and the titles there are short enough,
	as these are used by many visualisation tools.
	There are a few things to keep in mind when filling in the phenodata file, 
	so it might be advisable to check out our 
	<a href= "https://www.youtube.com/watch?v=eptYuRs2qws">video tutorial</a> (3 min)
	and/or <a href= "http://chipster.csc.fi/manual/visualisation-phenodata.html">the manual page</a> on how to fill in the phenodata file
	<br>
	 Tool (under Utilities category): 
	 <a href= "http://chipster.csc.fi/manual/ngs-create-experiment.html">Define NGS experiment </a></li>
<br>
	<b>HINT:</b> If your FASTQ and/or BAM files are big, at this point it might be wise to make sure your session is saved,
	then remove the large files from your session, and save the session again with another name. 
	It makes your session smaller and easier to handle! 
	You can always return to the earlier session with the bigger files if you notice a need for that later on.
	Note that the cloud sessions are not stored forever, so for longer storage, save the sessions (also) locally! 
	If the sessions are huge, it might be a good idea to store them in <a href= "https://www.youtube.com/watch?v=riipGwF-Qxs">Taito</a>.
	<br>  <br>
<li><b>Experiment level quality control (for the count table): </b><br>	
	Now we have our data in one table, and it is time to do some experiment level quality control 
	-this is the exciting part where you get to see whether your samples really are expressing differently! <br>
	<b>Note:</b> the tool expects you to use raw counts (in the count table), so don't do any normalisation!
	 <br>
	Check out from the PCA plot and heatmap that the samples are clustered as you would imagine them to cluster 
	-controls in one clump and so on- and that there are no outliers. <br>
	Now you can also see if there are some possible batch effects lurking in your data (see the Drosophila example session to see what you should be looking for.
	If you notice something, make sure you take it into account in the statistical testing / modeling phase!
	Tool (under Quality control category): 
	 <a href= "http://chipster.csc.fi/manual/deseq2-pca-heatmap.html">PCA and heatmap of samples using DESeq2) </a> </li> 
<br>
<li><b>Differential expression analysis (for the count table): </b><br>
<br>	
<br> KEHTAISKO PYYTAA SEIJALTA LYHYTTA VIDEOTA TAI NOTEBOOKIA AIHEESTA? TASTA JA SIITA linear modeling?
<br>
	Time for statistics! It is actually fairly tricky thing to estimate the differential expression due to a couple of things:
	we are testing usually thousands or tens of thousands of genes (multiple testing correction is needed), the expression values are not normally distributed
	 (instead, negative binomial distributions and generalised linear modeling are used) and 
	 the range in which a genes expression values vary varies from gene to gene (dispersion estimation). 
	 Luckily, there are tools that do all these tricky things for you. <br>
	 These tools are presented in our video tutorial: <br>
	 <a href= "https://www.youtube.com/watch?v=kPDy4FYF1yY&t=16s&list=PLjiXAZO27elABzLA0aHKS9chVA2TldoPF&index=1"> Differential expression analysis tools for RNA-seq tutorial video</a> (3 min)<br>
	 Things get a bit trickier if you have multiple variables, like treatment, gender, batch to take into account simultaniously. 
	 For these cases, you need to use the tool called EdgeR for multivariate experiments <br>
	 -check out the video tutorial for that 
	 <a href= "https://www.youtube.com/watch?v=gJnu3xUQ7ww&list=PLjiXAZO27elABzLA0aHKS9chVA2TldoPF&index=3"> here</a> (6 min), <br>
	 and a specific tutorial for cases where you need to use this "nested" option
	 <a href= "https://www.youtube.com/watch?v=82G_yMkPue4&t=11s&list=PLjiXAZO27elABzLA0aHKS9chVA2TldoPF&index=2">here</a> (4 min). <br>
	<br><br> IF NONE OF THESE OPTIONS WORK FOR YOUR DATA, THEN....??? Oliks joku kayta omaa matriisii tyokalu?? Missa??<br><br>
	 <b>Note:</b> the tools expect raw counts (in the count table) as input, so DON'T do any normalisation and don't use FPKM values or similar!
	 <br>
	Tools (under RNA-seq category): 
		 <a href= "http://chipster.csc.fi/manual/deseq2.html">DESeq2</a>, 
		 <a href= "http://chipster.csc.fi/manual/ngs-dea-edger-RNA.html">edgeR</a>,
		 <a href= "http://chipster.csc.fi/manual/ngs-dea-edger-RNA.html">edgeR for multivariate analysis</a>	
		 <a href= " http://chipster.csc.fi/manual/DEXSeq.html">DEXSeq</a>		 
		</li>
<br>
<li><b>Visualising your data: </b><br>	
<a href= "https://www.youtube.com/watch?v=PzZQVVD812w">Video tutorial for interactive visualisations</a> (7 min)
		</li> 
</ol>
<img src="RNA-seq-basic-steps.png"> <br><br>

</p>

</body>

