<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Seurat -Clustering </h2>

<h3>Description</h3>

<p>
This tool clusters the cells, does non-linear dimensional reduction and finds markers for the clusters.
</p>
 
 <h3>Parameters</h3>
<ul>
	<li>Number of principal components to use [10]</li>
	<li>Resolution for granularity [0.6]</li>
	<li>Cluster biomarker gene has to be expressed in at least this fraction of cells [0.25]</li>
	<li>Differential expression threshold for a cluster biomarker gene [0.25]</li>
	<li>Which test to use for finding marker genes [bimod, roc, t, tobit, poisson, negbinom]</li>
	<li>Only positive changes [TRUE]</li>
</ul>

 <h3>Details</h3>
 
<p>
As input, give the Seurat R-object (Robj) from the Seurat PCA -tool.
<br><br>
Seurat function <i>FindClusters</i> first constructs a KNN graph based on the Euclidean distance in PCA space,
and refines the edge weights between any two cells based on the shared overlap in their local neighborhoods (Jaccards distance).
To cluster the cells, the function uses the smart local moving algorithm to iteratively group cell groupings together
with the goal of optimizing the standard modularity function.
(See the links below for more information).
<br>
The <b>resolution</b> parameter sets the 'granularity' of the downstream clustering, 
with increased values leading to a greater number of clusters. 
We find that setting this parameter between 0.6-1.2 typically returns good results for single cell datasets 
of around 3K cells. Optimal resolution often increases for larger datasets 
-use a value above (below) 1.0 if you want to obtain a larger (smaller) number of communities.
<br>
The <b>principal components to use</b> need to be decided by the user based on the PCAplots from the Seurat PCA -tool.
</p>

<p>
Non-linear dimensional reduction (tSNE) is used for visualizing and exploring the dataset. Cells within the graph-based 
cluster determined above should co-localize on the tSNE plot -this is because the tSNE aims to place cells with similar 
local neighborhoods in high-dimensional space together in low-dimensional space. 
Same user-defined PCs are used here as with the clustering step.
The tSNE result is visualized in the <b>tSNEplot.pdf</b>.
</p>

<p>
Next, Seurat function <i>FindAllMarkers</i> is used to identify the positive and negative markers for the clusters compared
to all other cells.
The <b>min.pct</b> parameter requires a gene to be detected at a minimum percentage in either of the two groups of cells, 
and the <b>thresh.test</b> parameter requires a gene to be differentially expressed (on average) by some amount 
between the two groups. Both of these parameters can be set to 0, but with a dramatic increase in time - 
since this will test a large number of genes that are unlikely to be highly discriminatory. 
The marker genes for each recognized cluster are written in the <b>markers.tsv</b> file. 
<br>By default only positive markers are listed.
<br>For testing, Seurat currently implements "bimod" (likelihood-ratio test for single cell gene expression, McDavid et al., Bioinformatics, 2011, default), 
"roc" (standard AUC classifier), "t" (Students t-test), 
and "tobit" (Tobit-test for differential gene expression, as in Trapnell et al., Nature Biotech, 2014), 
"poisson", and "negbinom". 
The latter two options should only be used on UMI datasets, and assume an underlying poisson or negative-binomial distribution.
</p>

<p>
To get the markers for a specific cluster, you can use the tool <a href="/manual/ngs-filter-results-column.html"></a>Utilities / Filter table by column value</a> . 
For example, to get the markers for cluster 2, fill in the parameters accordingly: <br>
Column to filter by = cluster <br>
Does the first column have a title = no <br>
Cutoff = 2 <br>
Filtering criteria = equal-to  <br>
</p>
<p>
<p>
For more details, please check:<br>
The <a href="http://satijalab.org/seurat/get_started.html">Seurat tutorials</a>
<br><br>
The Seurat clustering approach was heavily inspired by these manuscripts 
which applied graph-based clustering approaches to scRNA-seq data:<br>
<a href="http://bioinformatics.oxfordjournals.org/content/early/2015/02/10/bioinformatics.btv088.abstract">[SNN-Cliq, Xu and Su, Bioinformatics, 2015] </a> 
and 
<br>CyTOF data
<a href="http://www.ncbi.nlm.nih.gov/pubmed/26095251">[PhenoGraph, Levine et al., Cell, 2015]</a>.
<br><br>
Smart local moving algorithm:<br>
<a href="http://dx.doi.org/10.1088/1742-5468/2008/10/P10008">[SLM, Blondel et al., Journal of Statistical Mechanics]</a>

</p>

<h3>Output</h3>

<ul>
<li>seurat_obj.Robj: The Seurat R-object to pass to the next Seurat tool, or to import to R. Not viewable in Chipster. </li>
<li>tSNEplot.pdf: Quality control plots (explained in detail above) </li>
<li>markers.tsv: Top markers</li>
</ul>



</body>

