<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Seurat -Setup and QC </h2>

<h3>Description</h3>

<p>
This tool can be used to setup the Seurat object from two possible inputs: 
<ol>
<li>a .tar package of a folder which contains the <b>10X Genomics</b> output files (see more info below)  </li>
<li>the DGE table (from <b>DropSeq</b>) -this can be generated with the <i>Digital expression </i> tool </li>
</ol>
After setting up the R-object (.Robj), some quality control plots are drawn.
</p>
 
 <h3>Parameters</h3>
<ul>
		<li>Project name for plotting [Project_name]</li>
		<li>Include genes with detected expression in at least this many cells [3]</li>
		<li>Include cells where at least this many genes are detected [200]</li>
		<li>Perform log normalization [T]</li>
		<li>Scale factor in the log normalization [10000]</li>		
</ul>
	
	
 <h3>Details</h3>
  
<p>
As inputs, user can give <b>EITHER</b>
<br> a .tar packed <b>folder</b> of 10X Genomics files output files (barcodes.tsv, genes.tsv and matrix.mtx 
-these are generated by the Cell Ranger -pipeline, and can be found under a folder corresponding to the hg19-folder in this 
<a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/matrices">example</a>), 
<br><b>OR</b> 
<br>a (DropSeq) digital gene expression  (DGE) matrix (which one can generate using <i>Digital expression </i> tool).
<br><b>Make sure that your input is assigned correctly (under the parameters).</b>
</p>

<p>
After setting up the object, some basic filtering is performed. 
At this step, user can define:
<ul>
	<li>Filtering genes: in how many cells a gene is required to be detected </li>
	<li>Filtering cells: how many genes are required to be detected in a cell </li>
	<li>Log normalization: whether to perform log normalization to data (default: yes) </li>
	<li>Scaling for log normalization: scale each cell to this number of molecules before log transformation (default: 10000) </li>
</ul>
</p>

<p>
Next, some quality control plots are drawn in the QC-plots.pdf file. 
Note that it might be easier to view this large pdf with the external viewer.
<br>
The first plot shows the number of genes (nGene), UMIs (nUMI) and 
the percentage of mitochondrial genes (percent.mito) in the cells.
The next plot compares the percent.mito and nGene to nUMI.
These plots can be used to estimate the outliers: the upper limits for number of genes per cell and mitochondrial percentage.
Check the plots and determine these cut-offs, and use them in the filtering & regression tool.
</p>


<p>
For more details, please check the <a href="http://satijalab.org/seurat/get_started.html">Seurat tutorials</a>.
</p>

<h3>Output</h3>

<ul>

<li>seurat_obj.Robj: The Seurat R-object to pass to the next Seurat tool, or to import to R. Not viewable in Chipster. </li>
<li>QCplots.pdf: Quality control plots (explained in detail above) </li>
</ul>



</body>

