<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>Statistics / Two group tests</h2>

<h3>Description</h3>

Tests every gene's expression for a difference between two groups.

<h3>Parameters</h3>

<ul>
<li>Column with group information () [] </li>
<li>Column with pairing information () [] </li>
<li>Test (empiricalBayes, t-test, fast t-test, F-test, Mann-Whitney, LPE, RankProd) [empirical Bayes] </li>
<li>p-value adjustment method (none, Bonferroni, Holm, Hochberg, BH, BY) [BH]</li>
<li>P-value threshold (0...1) [0.05]</li>
<li>Include genes whose p-value is NA (yes, no) [no]</li>
</ul>

<h3>Details</h3>

<p>Runs the specified test, and returns the genes that have a corrected p-value smaller than the specified 
p-value cut-off. If you want to get a p-value for every single gene in your data set, set the p-value cut-off to 1. This
should return a new gene list of equal length to the original one.</p>

<p> Mark both groups in the phenodata file with numbers, and use smaller number for the control/baseline group. 
So for example control samples can be coded with "1" and treatment samples with "2" in the group column. 
Then Chipster knows to compare treatment to control (control is coded with a smaller number than the treatment). 
Since this makes a difference when interpreting the results such as fold change, it is worth paying attention to.</p>

<p>The test empiricalBayes is essentially a t-test where variance estimates are shrank towards a pooled estimate, 
resulting in far more stable inference when the number of samples is small. The empiricalBayes test
is also much faster than the usual t-test. LPE test is especially well suited for small sample sizes. F-test compares the differences in variance between the two 
groups, while the other tests compare the means.</p>

<p>If you would like to perform a paired test (t, Mann-Whitney or empiricalBayes), you need to describe the paired samples with the same number in the phenodata file. Please note that if a sample does not have a pair, it is not used 
in assessing the significance.
</p>

<p>Multiple testing correction options are Bonferroni, Holm, and Hochberg for family-wise error rate (FWER), and
Benjamini-Hochberg and Benjamini-Yakutieri for false discovery rate (FDR). Of these Bonferroni is the most
conservative, and the FDR-based adjustments are less conservative.</p>

<h3>Output</h3>

<p>A text file containing the gene expression values and the p-value for the test. </p>


<h3>References</h3>

<p>This tool uses Bioconductor packages limma and LPE. Please cite the following article, if you used empirical Bayes
from limma package:</p>

<p>Smyth, G. K. (2004). Linear models and empirical Bayes methods for assessing
differential expression in microarray experiments. Statistical Applications in Genetics and Molecular Biology, Vol. 3, No. 1, Article 3.</p>

<p>Cite the following articles, if you used the LPE test:</p>

<p>Jain et. al. Local-pooled-error test for identifying differentially expressed genes with
a small number of replicated microarrays, Bioinformatics, 2003, Vol 19, No. 15, pp:
1945-1951.</p>

<p>Jain et. al. Rank-invariant resampling based estimation of false discovery rate for analysis
of small sample microarray data, BMC Bioinformatics, 2005, Vol 6, 187.</p>


</body>

