<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>VirusDetect with own genome</h2>

<h3>Description</h3>

<p>This tool runs the VirusDetect pipeline, that performs virus identification using sRNA sequencing data. 
Given a FASTQ file, it performs <i>de novo</i> assembly and reference-guided assembly by 
aligning sRNA reads to the reference database of known viruses. The assembled contigs are 
compared to the reference virus sequences for virus identification.</p>

<p>More detailed definition of Virus detect pipeline can be found from 
the <a href="http://bioinfo.bti.cornell.edu/cgi-bin/virusdetect/index.cgi">
home page of VirusDetect</a>.</p>

<p> If possible, the input data should be cleaned from sequences originating from the genome of the host organism.
This can be done by mapping the query sequences against the genome of the host organisms and selecting 
only those reads that do not match to the host genome.<p>

<p>If the host genonome is not available or it is available in Chipter, you should use the
<i>VirusDetect</i> tool in stead of this tool.


<p>If the genome of the host organism is not available in Chipster, but you have the host genome 
as a fasta formatted sequence file, you can use this tool to automatically calculate the required 
BWA indexes and the host genome filtering for your input data. You should note that it may take several 
hours to calculate BWA indexes for a large genome.<p>

<p>When the VirusDetect analysis is finished, the BWA indexes of the host genome are returned as 
one tar-formatted archive file. This archive file can be used as input file, in stead of the fasta formatted genome,  for succeeding 
DirusDetect and BWA jobs so that you don't have to repeat the time consuming indexing process.<p>


<h3>Input Files</h3>
This tool requires two input files:
<ul>
<li><b>Input data (the reads)</b> should be given as FASTQ formatted sequence file.</li>
<li><b>Host genome</b> should be given as fasta formatted sequence file or as tar formatted BWA index file from previous VirusDetect job.</li>
</p>
</ul>

<h3>Parameters</h3>
<ul>
<p>
<li><b>Reference virus database</b> VirusDetect is mainly used for detecting plant viruses, 
but you can use it for other viruses too. Use this parameter to select a virus reference database 
matching you virus type.</li>
<li><b>Reference virus coverage cuttoff</b> Coverage cutoff of a reported virus contig by reference virus sequences.</li>
<li><b>Assembled virus contig cuttoff</b> Coverage cutoff of a reported virus reference sequences by assembled virus contigs.</li>
<li><b>Depth cutoff</b> Depth cutoff of a reported virus reference</li>
</p>
</ul>
<h3>Output</h3>

VirusDetect produces large amount of different files as reports. Output related options are used to 
select, what data is returned. By default VirusDtetect returns following files:
<ul>
<li><b>virusdetect_contigs.fa</b> Sequences of non-redundant contigs derived through reference-guided and de novo assemblies.</li>
<li><b>contig_sequences.undetermined.fa</b> Sequences of contigs that do not match to virus references.</li> 
<li><b>blastn_matching_refrences.html</b> files listing reference viruses that have corresponding virus contigs identified by BLASTN. A pdf formatted report file is returned for each match</li>
<li><b>blastn_matches.tsv</b> a table of blastn matches to the reference virus database.</li>
<li><b>blastx_matching_refrences.html</b> files listing reference viruses that have corresponding virus contigs identified by BLASTX. A pdf formatted report file is returned for each match </li>
<li><b>blastx_matches.tsv</b> a table of blastn matches to the reference virus database.</li>
<li><b>hostgenome_bwa_index.tar</b> BWA indexes for the given host genome.</li> 
</ul>

If parameter <b>Return matching reference sequences</b> is turned on the also following files are returned 
<ul>
<li><b>blastn_matching_references.fa</b> and <b>.fai</b>. Virus reference sequences that produced hits for blastn search with the potential virus contigs.</lI>
<li><b>blastn_matching_references.fa</b> and <b>.fai</b>. Virus reference sequences that produced hits for blastn search with the potential virus contigs.</lI>
</li>
</ul>
If parameter <b>Return matching reference sequences</b> is turned on the also following files are returned 
<ul>
<li><b>blastn_matches.bam</b> and <b>.bai</b>. BAM file containing the blastn alignment of each contig to its corresponding virus reference sequences.</li> 
<li><b>blastx_matches.bam</b> and <b>.bai</b>. BAM file containing the blastx alignment of each contig to its corresponding virus reference sequences.</li> 
</ul>
<p>
Note: If you select both the blastn_matching_references.fa + .fai and blastn_matches.bam + .bai, 
(or the corresponding blastx files) you can use the Genome Browser to visualize the BLAST reulst.
In the Genome Browser the <i>blastn_matching_references.fa</i> should be assigned to be used as the genome.
Each reference virus sequence is then listed in the <i>Chromosome</i> pull down menu.  
</p>

<p>If parameter <b>Return results in one archive file</b> is selected, the all the outputfiles are stored to
a single tar formatted output file. This feature is useful if you run VirusDetetc to several input files in 
the same time. The tar formatted output file can be expanded with tool <i>Extract .tar.gz file</i>.




</body>

