<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head><title></title></head>
<body>

<h2>VirusDetect with own genome</h2>

<h3>Description</h3>

<p>This tool runs the VirusDetect pipeline, that performs virus identification using small RNA (sRNA) sequencing data. 
Given a FASTQ file, it performs <i>de novo</i> assembly and reference-guided assembly by 
aligning sRNA reads to the reference database of known viruses. The assembled contigs are 
compared to the reference virus sequences for virus identification.</p>

<p>More detailed definition of Virus detect pipeline can be found at 
the <a href="http://bioinfo.bti.cornell.edu/cgi-bin/virusdetect/index.cgi">
home page of VirusDetect</a>.</p>

<p> If possible, the input data should be cleaned from sequences originating from the host genome.
This can be done by mapping the reads against the host genome and selecting 
only those reads that did not match. If the host genome is not available in Chipster, but you have the host genome 
as a fasta formatted sequence file, you can use this tool to automatically calculate the required 
BWA indexes and perform the host genome filtering for your data. Please note that it may take several 
hours to calculate BWA indexes for a large genome.<p>

<p>When the VirusDetect analysis is finished, the BWA indexes of the host genome are returned as 
one tar-formatted archive file. This archive file can be used as an input file, instead of the fasta formatted genome, for the subsequent 
VirusDetect and BWA jobs so that you don't have to repeat the time consuming indexing process.<p>


<h3>Input Files</h3>
This tool requires two input files:
<ul>
<li><b>sRNA reads</b> should be given as FASTQ formatted sequence file.</li>
<li><b>Host genome</b> should be given as fasta formatted sequence file or as tar formatted BWA index file from previous VirusDetect job.</li>
</p>
</ul>

<h3>Parameters</h3>
<ul>
<p>
<li><b>Reference virus database</b> VirusDetect is mainly used for detecting plant viruses, 
but you can use it for other viruses too. Use this parameter to select a virus reference database 
matching you virus type.</li>
 </p>
 <p>
<li><b>Reference virus coverage cutoff</b>: The match between the virus reference and 
 the contig sequence must cover at least this proportion of the contig in order 
 to make the match significant for the virus assignment. For example the default value 
 0.75 means that only contigs that match to the reference viruses for more than 75% of their length
 are considered as significant matches. </li>
 </p>
<p>
 <li><b>Assembled virus contig cutoff</b>: The virus assignment is reported only if more than the given proportion of the reference genome is
 covered with significant matches to the generated contigs. For example, the default value of 0.1 means that at least 10 % 
 of the reference virus genome must match to the contigs generated by VirusDetect.</li>
 </p>
<p>
 <li><b>Depth cutoff</b>: Depth means the average total number of times each nucleotide of 
 the reference sequence is covered by sequences identified in the sample.</li>
</p>
</ul>
<h3>Output</h3>

VirusDetect produces several result files. Output related parameters are used to 
select what data is returned. By default VirusDetect returns the following files:
<ul>
<li><b>virusdetect_contigs.fa</b> Sequences of non-redundant contigs derived through reference-guided and de novo assemblies.</li>
<li><b>contig_sequences.undetermined.fa</b> Sequences of contigs that do not match to virus references.</li> 
<li><b>blastn_matching_refrences.html</b> files listing reference viruses that have corresponding virus contigs identified by BLASTN. A pdf formatted report file is returned for each match</li>
<li><b>blastn_matches.tsv</b> a table of blastn matches to the reference virus database.</li>
<li><b>virusderect_matches_blastn.fa</b>  Sequences of contigs that match to virus references by BLASTN.</li>
<li><b>blastx_matching_refrences.html</b> files listing reference viruses that have corresponding virus contigs identified by BLASTX. A pdf formatted report file is returned for each match </li>
<li><b>blastx_matches.tsv</b> a table of blastn matches to the reference virus database.</li>
<li><b>virusderect_matches_blastx.fa</b>  Sequences of contigs that match to virus references by BLASTX.</li>
<li><b>hostgenome_bwa_index.tar</b> BWA indexes for the given host genome.</li>
<li><b>undetermined.html</b> A table of listing the length, siRNA size distribution and 21-22nt percentage of undetermined contigs. Potential virus contigs (21-22 nt > 40%) are indicated in green.</li>
</ul>

If the parameter <b>Return matching reference sequences</b> is turned on, also the following files are returned 
<ul>
<li><b>blastn_matching_references.fa</b> and <b>.fai</b>. Virus reference sequences that produced hits for blastn search with the potential virus contigs.</lI>
<li><b>blastn_matching_references.fa</b> and <b>.fai</b>. Virus reference sequences that produced hits for blastn search with the potential virus contigs.</lI>
</li>
</ul>
If the parameter <b>Return matching reference sequences</b> is turned on, also the following files are returned 
<ul>
<li><b>blastn_matches.bam</b> and <b>.bai</b>. BAM file containing the blastn alignment of each contig to its corresponding virus reference sequences.</li> 
<li><b>blastx_matches.bam</b> and <b>.bai</b>. BAM file containing the blastx alignment of each contig to its corresponding virus reference sequences.</li> 
</ul>
<p>
Note: If you select both the blastn_matching_references.fa + .fai and blastn_matches.bam, 
(or the corresponding blastx files) you can use the Genome Browser to visualize the BLAST result.
In the Genome Browser the <i>blastn_matching_references.fa</i> should be assigned to be used as the genome.
Each reference virus sequence is then listed in the <i>Chromosome</i> pull down menu.  
</p>

<p>If the parameter <b>Return results in one archive file</b> is selected, then all the output files are stored to
a single tar package file. This feature is useful if you run VirusDetect on several input files at 
the same time. The tar file can be expanded with tool <i>Extract .tar.gz file</i>.




</body>

